lib/functoria.mli

```diff
@@ -192,28 +225,29 @@ class type ['ty] configurable = object
   (** [module_name] is the name of the module implementing the
       configurable. *)
 
-  method packages: string list value
+  method packages: package list value
   (** [packages] is the list of OPAM packages which needs to be
       installed before compiling the configurable. *)
 
-  method libraries: string list value
-  (** [libraries] is the list of OCamlfind libraries to include and
-      link with the configurable. *)
-
   method connect: Info.t -> string -> string list -> string
   (** [connect info mod args] is the code to execute in order to
       initialize the state associated with the module [mod] (usually
       calling [mod.connect]) with the arguments [args], in the context
       of the project information [info]. *)
 
-  method configure: Info.t -> (unit, string) Rresult.result
+  method configure: Info.t -> (unit, Rresult.R.msg) Rresult.result
   (** [configure info] is the code to execute in order to configure
       the device. This might involve generating more OCaml code,
-      running bash scripts, etc. *)
+      running shell scripts, etc. *)
+
+  method build: Info.t -> (unit, Rresult.R.msg) Rresult.result
+  (** [build info] is the code to execute in order to build
+      the device. This might involve generating OCaml code,
+      running shell scripts, etc. *)
```

Comment:
There is a clear overlap between the function of configure and build, according to the documentation. It should be much clearer what should go where and when things are executed.